export_schema.py

```diff
@@ -0,0 +1,279 @@
+from dataclasses import dataclass
+from enum import auto, Enum
+from typing import List, Union
+
+################################################################################
+# Following section is the defining a few utility data structures
+
+# As many of the serialization library doesn't have the concept of map
+# We use a list of sting-string pair to represent a map<string, string>
+@dataclass
+class StringStringPair:
+    key: str
+    value: str
+
+################################################################################
+# Following section is the defining the permissible argument types for operators
+
+# Copied from torchgen/model.py
+# !!! Consider if we want to add aliased dtypes, e.g float == float32. (Sherlock: No!)
+class ScalarType(Enum):
+    Byte = auto()
+    Char = auto()
+    Short = auto()
+    Int = auto()
+    Long = auto()
+    Half = auto()
+    Float = auto()
+    Double = auto()
+    ComplexHalf = auto()
+    ComplexFloat = auto()
+    ComplexDouble = auto()
+    Bool = auto()
+    BFloat16 = auto()
+
+# Copied from torch/_C/__init__.pyi.in
+# !!! Consider if we want to keep all of them, as we figures out what to do with SparseTensor
+class Layout(Enum):
+    # Defined in torch/csrc/utils/tensor_layouts.cpp
+    strided = auto()
+    sparse_coo = auto()
+    sparse_csr = auto()
+    sparse_csc = auto()
+    sparse_bsr = auto()
+    sparse_bsc = auto()
+    _mkldnn = auto()
+
+
+# Copied from torch/_C/__init__.pyi.in
+class MemoryFormat(Enum):
+    # Defined in torch/csrc/utils/tensor_memoryformats.cpp
+    contiguous_format = auto()
+    channels_last = auto()
+    channels_last_3d = auto()
+
+    # !!! This is annoying! preserve_format cannot be used as a tensor propery, as it can only used as an argument for operators
+    preserve_format = auto()
+
+
+# Copied from torch/_C/__init__.pyi.in
+@dataclass
+class Device:
+    # Defined in torch/csrc/Device.cpp
+    type: str
+    index: int
+
+
+SymInt = Union[
+    int,
+    str
+]
+
+Scalar = Union[int, float, bool]
+
+
+# ArgumentType = Union[
```

Comment:
Seems we are missing operator definition? Otherwise this `ArgumentType` won't be used anywhere.